### PR TITLE
Gazette detail page

### PR DESCRIPTION
**Pull request by AliciadW**
Added content and basic styling for Gazette Detail Page.
Still need to figure out:
1) How to get rid of the Title (top)
2) How to add href to button link

Screenshot 2020-05-11 at 13 57 11

### CODE REVIEW

File: web/sites/default/services.yml [hunk 1]
```diff
@@ -0,0 +1,174 @@
+parameters:
```
Comment (longhotsummer) [inline web/sites/default/services.yml:1]: This file shouldn't be committed.

File: web/sites/default/settings.php [hunk 2]
```diff
@@ -805,3 +805,14 @@
 $config_directories['sync'] = '../config/default';
 
 $settings['node_title_length_chars'] = 1023;
+$databases['default']['default'] = array (
```
Comment (longhotsummer) [inline web/sites/default/settings.php:808]: Rather copy this file into `web/sites/default/settings.local.php` and make these changes there.

File: web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * - logged_in: Flag for authenticated user status. Will be true when the
+ *   current user is a logged-in member.
+ * - is_admin: Flag for admin user status. Will be true when the current user
+ *   is an administrator.
+ *
+ * @see template_preprocess_node()
+ */
+#}
+{{ attach_library('bootstrap_barrio/node') }}
+{{ attach_library('liibarrio/polyfill-sticky') }}
```
Comment (longhotsummer) [inline web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig:63]: This can be removed, we don't need the sticky polyfill.

File: web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    'node--type-' ~ node.bundle|clean_class,
+    node.isPromoted() ? 'node--promoted',
+    node.isSticky() ? 'node--sticky',
+    not node.isPublished() ? 'node--unpublished',
+    view_mode ? 'node--view-mode-' ~ view_mode|clean_class,
+    'clearfix',
+  ]
+%}
+<article{{ attributes.addClass(classes) }}>
+  <div class="col-12">
```
Comment (longhotsummer) [inline web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig:77]: I don't think this is necessary, and without a `row` parent, the margins become weird.

**Comment by longhotsummer** [comment]
@AliciadW maybe to simplify the title issue we just use the main page title, and put the date under that? So the node template doesn't worry about rendering the title? But I'm not sure how it's getting there in the first place.

File: web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+%}
+{%
+  set title_classes = [
+  'field__label',
+  label_display == 'visually_hidden' ? 'visually-hidden',
+]
+%}
+{# Not sure how to correclty add href #}
+<button class="btn btn-primary" href="{{ element['#url_title'] }}">
+  {{ element['#title'] }}
```
Comment (longhotsummer) [inline web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig:56]: Try `items[0].content`?

Comment (AliciadW) [reply web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig:56]: The button text displays as it should with `{{ element['#title'] }}`. 
According to what I've read the `href` should be done the same, although it doesn't work for some reason. 
I'll keep looking into it.

**Comment by AliciadW** [comment]
> @AliciadW maybe to simplify the title issue we just use the main page title, and put the date under that? So the node template doesn't worry about rendering the title? But I'm not sure how it's getting there in the first place.

I will remove the second title that I added into the template, but feel that I need to understand where the first one is coming from, to know how to handle this in the future.

**Comment by longhotsummer** [comment]
> > @AliciadW maybe to simplify the title issue we just use the main page title, and put the date under that? So the node template doesn't worry about rendering the title? But I'm not sure how it's getting there in the first place.
> 
> I will remove the second title that I added into the template, but feel that I need to understand where the first one is coming from, to know how to handle this in the future.

Yes, that's my problem too! Let me know when you find out.

File: web/sites/default/services.yml (hunk 1 above)
Comment (AliciadW) [reply web/sites/default/services.yml:1]: Removed

File: web/sites/default/settings.php (hunk 2 above)
Comment (AliciadW) [reply web/sites/default/settings.php:808]: Done

Comment (AliciadW) [reply web/sites/default/settings.php:808]: Done

File: web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig (hunk 3 above)
Comment (AliciadW) [reply web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig:63]: Done

File: web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig (hunk 4 above)
Comment (AliciadW) [reply web/themes/custom/liibarrio/templates/node--government-gazette--full.html.twig:77]: Sorry, that was an oversight.

File: web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig (hunk 5 above)
Comment (AliciadW) [reply web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig]: Fixed this

**Comment by AliciadW** [comment]
> > > @AliciadW maybe to simplify the title issue we just use the main page title, and put the date under that? So the node template doesn't worry about rendering the title? But I'm not sure how it's getting there in the first place.
> > 
> > 
> > I will remove the second title that I added into the template, but feel that I need to understand where the first one is coming from, to know how to handle this in the future.
> 
> Yes, that's my problem too! Let me know when you find out.

Still trying to figure this out, but removed the second title for now.

**Comment by AliciadW** [comment]
@longhotsummer I have addressed the comments. Please review again. Thanks!

File: web/themes/custom/liibarrio/templates/node--legislation--full.html.twig [hunk 6]
```diff
@@ -119,22 +119,24 @@
...
     <div class="tab-pane fade" id="history" role="tabpanel" aria-labelledby="history-tab">
       {{ content.field_lifecycle_json }}
     </div>
-  </div> 
+  </div>
 
 
 </article>
+
+<script>console.log({{ _context | json_encode | raw }});</script>
```
Comment (longhotsummer) [inline web/themes/custom/liibarrio/templates/node--legislation--full.html.twig:142]: Don't forget to remove this :)

Comment (AliciadW) [reply web/themes/custom/liibarrio/templates/node--legislation--full.html.twig]: 🤦 
Removed :-)

File: web/sites/default/services.yml (hunk 1 above)
Comment (longhotsummer) [reply web/sites/default/services.yml:1]: It's still there... ?

File: web/sites/default/settings.php [hunk 7]
```diff
@@ -803,5 +803,3 @@
   include $app_root . '/' . $site_path . '/settings.local.php';
 }
 $config_directories['sync'] = '../config/default';
-
```
Comment (longhotsummer) [inline web/sites/default/settings.php:806]: I think we may need to leave this in.

Comment (AliciadW) [reply web/sites/default/settings.php]: Added it back.

**Comment by AliciadW** [comment]
Screenshot 2020-05-12 at 14 46 13

**Comment by longhotsummer** [comment]
Fixes #63

File: .gitignore [hunk 8]
```diff
@@ -25,3 +25,7 @@ web/update.php
 /vendor/
 /web/modules/contrib/
 /web/themes/contrib/
+
+/web/sites/default/services.yml
```
Comment (cristiroma) [inline .gitignore:29]: We don't exactly understand why this change is part of this task and why it ignores a Drupal file.

File: web/core/modules/config/tests/src/Functional/ConfigEntityListTest.php [hunk 9]
```diff
@@ -50,7 +50,7 @@ public function testList() {
 
     // Get a list of ConfigTest entities and confirm that it contains the
     // ConfigTest entity provided by the config_test module.
-    // @see config_test.dynamic.dotted.default.yml
+    // @see config_test.dynamic.dotted.services.yml
```
Comment (cristiroma) [inline web/core/modules/config/tests/src/Functional/ConfigEntityListTest.php:53]: Is this change intentional? It is part of Drupal core, but not core update took place.

File: web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+{%
+  set title_classes = [
+  'field__label',
+  label_display == 'visually_hidden' ? 'visually-hidden',
+]
+%}
+
+<a class="btn btn-primary my-3" href="{{ element[0]['#url_title'] }}" target="_blank">
+  Download PDF
+</a>
```
Comment (cristiroma) [inline web/themes/custom/liibarrio/templates/field--node--field-external-link--government-gazette.html.twig:57]: I think this template wasn't meant for `field_external_link` but rather for `field_file`. Also, while overriding a field template can work, creating a custom field formatter is the correct solution because:

- creates a reusable component
- if you override the template and the field formatter is changed in any way, there is a chance that the URL to the file is no longer `{{ element[0]['#url_title'] }}`, introducing a bug.

However, if you do stick with this solution, use `{{ "Download" | t }}` so that the string gets translated.

**Comment by cristiroma** [comment]
Looking at the prototype, the best option to use the Display Suite (https://www.drupal.org/project/ds) module then create a single general template (with several regions) for the whole website. The content region could be split like this:

```
[       Top left  |    Top right    ]
[             Full width            ]
[   Content      ]|[    Sidebar     ]
```

Then each region can contain fields, views etc.

There's no need to make a twig for each node type, and is especially advisable to avoid this behavior as much as possible.

With this kind of setup, managing display of content type pages can be achieved by using the drag and drop functionality to manage field display in region from the "Manage display" back-end screen. And CSS for styiling.
